app/helpers/application_helper.rb

```diff
@@ -124,6 +124,13 @@ def make_line(row, td_opts)
   #
   def reload_with_args(new_args)
     uri = request.url.sub(/^\w+:\/+[^\/]+/, "")
```

Comment:
What about the following in place of lines 126-132 (so that `uri` is only ever defined once):
```
if request.url.include? "ajax"
  uri = request.url.sub(/ajax/, "observer")
else
  uri = request.url.sub(/^\w+:\/+[^\/]+/, "")
end
```